src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -67,6 +67,9 @@ public EETypeNode(TypeDesc type)
             Debug.Assert(!type.IsRuntimeDeterminedSubtype);
             _type = type;
             _optionalFieldsNode = new EETypeOptionalFieldsNode(this);
+
+            if (!type.IsGenericDefinition)
```

Comment:
We don't create an EEType for the base type of a generic definition though, so we shouldn't validate it.  I think a better way would be to move the `IsGenericDefinition` to the top of `CheckCanGenerateEEType` so we don't go down that path for defs.